apps/mymedicare_cb/tests/test_callback_slsx.py

```diff
@@ -434,3 +429,467 @@ def catchall(url, request):
+               with error text: "Found user's hicn did not match".
+
+            6. Restore saved_hicn_hash in Crosswalk prior to next test. Restore crosswalk state to same as #3.
+
+            7. The bene's MBI has been changed in the mock SLSx user_info response.
+               This response is mocked by:  MockUrlSLSxResponses.slsx_user_info_mock_changed_mbi
+
+            8. Restore saved_mbi_hash in Crosswalk prior to next test. Restore crosswalk state to same as #3.
+
+            9. The bene's HICN & MBI (both) have been changed in the mock SLSx user_info response.
+               This response is mocked by:  MockUrlSLSxResponses.slsx_user_info_mock_changed_hicn_mbi
+        '''
+        # create a state
+        state = generate_nonce()
+        AnonUserState.objects.create(
+            state=state,
+            next_uri="http://www.google.com?client_id=test&redirect_uri=test.com&response_type=token&state=test")
+
+        # mock fhir patient endpoint (back end bfd) with fhir_id == "-20140000008325"
+        @urlmatch(netloc="fhir.backend.bluebutton.hhsdevcloud.us", path="/v1/fhir/Patient/")
+        def fhir_patient_info_mock(url, request):
+            return {
+                "status_code": status.HTTP_200_OK,
+                "content": patient_response,
+            }
+
+        @all_requests
+        def catchall(url, request):
+            raise Exception(url)
+
+        # 1. First successful matching for beneficiary having valid only hicn and EMPTY mbi.
+        with HTTMock(MockUrlSLSxResponses.slsx_token_mock,
+                     MockUrlSLSxResponses.slsx_user_info_empty_mbi_mock,
+                     MockUrlSLSxResponses.slsx_health_ok_mock,
+                     MockUrlSLSxResponses.slsx_signout_ok_mock,
+                     fhir_patient_info_mock,
+                     catchall):
+            response = self.client.get(self.callback_url, data={"req_token": "test", "relay": state})
+            # assert http redirect
+            self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+
+        # Get crosswalk values.
+        cw = Crosswalk.objects.get(id=1)
+
+        # Assert correct crosswalk values:
+        self.assertEqual(cw.user.id, 1)
+        self.assertEqual(cw.user.username, "00112233-4455-6677-8899-aabbccddeeff")
+        self.assertEqual(cw.fhir_id, "-20140000008325")
+        self.assertEqual(cw._user_id_hash, "f7dd6b126d55a6c49f05987f4aab450deae3f990dcb5697875fd83cc61583948")
+        self.assertEqual(cw._user_mbi_hash, None)
+
+        # Validate ArchiveCrosswalk count
+        self.assertEqual(ArchivedCrosswalk.objects.count(), 0)
+
+        # Validate logging
+        log_list = self._get_log_lines_list("audit.request_logger")
+        self.assertEqual(len(log_list), 2)
+
+        #   Get last log line
+        log_dict = json.loads(log_list[len(log_list) - 1])
+
+        #   Set working copy of schema
+        log_schema = copy.deepcopy(MYMEDICARE_CB_GET_UPDATE_BENE_LOG_SCHEMA)
```

Comment:
Making use of a DEEP COPY of the schema so that the original schema can be used/reset as a basis for each of the tests. Changes to the `log_schema` dict do not affect the original `MYMEDICARE_CB_GET_UPDATE_BENE_LOG_SCHEMA` schema dict.